### DISCUSSION TITLE
Is using macro software allowed?

### DISCUSSION THREAD

**Discussion by brian7262** [chromiecraft/chromiecraft#6118] [General]
The manufacturer of my mouse recommends installing their official software for firmware updates and configuring the mouse which includes functionality to create and assign macros as seen at https://www.razer.com/synapse-3 . Am I allowed to use that on chromiecraft?

**Comment by r-o-b-o-t-o** [comment]
ChromieCraft forbids **automation of gameplay**. In a nutshell:
- Any action performed that does not originate from a player's **immediate physical action** is against the rules
- Triggering inputs based on a non-deterministic system is against the rules (e.g. pressing one key and having software reroute it to another key-press based on varying factors)
- Triggering inputs with a delay after the player's physical action is against the rules (e.g. pressing one key and having software send delayed or repeated key-presses)
- Triggering another input is allowed **as long as the other rules above are respected** (e.g. pressing one key and having software press another key at the exact same time)
